### DISCUSSION TITLE
ListensToLogcatMessages WSS support

### DISCUSSION THREAD

**Discussion by muhanov** [appium/java-client#2154] [Ideas]
Hi team,
I'm observing that ListensToLogcatMessages.startLogcatBroadcast(String host, int port); hardcodes inside the WS protocol.
endpointUri = new URI(String.format("ws://%s:%s/ws/session/%s/appium/device/logcat", host, port, ((RemoteWebDriver)this).getSessionId()));

For some purposes I use Appium environment remotely by HTTPS.
So, in this case connection is refusing by Appium server because I should use WSS instead.

I merely copy-pasted the method implementation to my project and changed WS to WSS and it works!
Please review the possibility to extend the client implementation to support WSS as well

Thanks, M

**Comment by valfirst** [comment]
Fixed in `9.3.0` (#2182)
